SMILE Fixes 

state.py
Python 2.7: xrange
Python 3+: range

Python 2.7: ---.next()
Python 3+: next(---)
Line 1410 - self.__child_iterator.next()._clone(self))
Line 1442 self.__child_iterator.next()._clone(self))

Python 2.7: __metaclass__ = form
Python 3+: class name(object, metaclass=form)
Line 170 __metaclass__ = StateClass

Python 2.7: xrange
Python 3+: range
*Slight differences in range implementation but shouldn't affect SMILE
Line 463 for distinguisher in xrange(256)

Python 2.7: .__new__(multiple arguments)
Python 3.6: __new__(one argument) (?)
*not sure about this one, need to double check
Line 181 return super(State, cls).__new__(cls, *pargs, **kwargs)

video.py
Python 2.7: dict.keys() returns the keys
Python 3+: dict.keys() returns a dictionary view: dict_keys[(keys)]
Line 341 self._constructor_param_names = param.keys() --> list(param) returns keys
Line 720 pos_prope = (xy_pos_props.keys() + ...)

main.py
Line 369 glVertexAttriPointer(0,2, GL_INT, GL_FALSE, 0, "\x00\x00\x00\x0a\x00\x00\x00\x0a")
Received error: File "C:\Users\limhe\OneDrive\School\UVa\CompAnalyLab\smile-master\smile - Copy\main.py", line 136, in _on_start
     self.do_flip(block=True)
   File "C:\Users\limhe\OneDrive\School\UVa\CompAnalyLab\smile-master\smile - Copy\main.py", line 370, in do_flip
     "\x00\x00\x00\x0a\x00\x00\x00\x0a")
   File "kivy\graphics\opengl.pyx", line 1545, in kivy.graphics.opengl.glVertexAttribPointer (kivy\graphics\opengl.c:19612)
 TypeError: Argument 'data' has incorrect type (expected bytes or int).

 StartScreen_2.py
 removed the **kwargs argument from __init__

 moving_dots.py
 removed **kwargs in super(type(self), self).__init__(**kwargs) in class _MovingDotsWidget(Widget)

changed xrange to range

	- able to run the program
	- dots disappears and exits after pressing 'shift' key